File: legacy/java/components/StrataCommon/src/strata1/common/producerconsumer/DisruptorExceptionHandler.java
// ##########################################################################
// # File Name:	DisruptorExceptionHandler.java
// #
// # Copyright:	2014, Sapientia Systems, LLC. All Rights Reserved.
// #
// # License:	This file is part of the StrataCommon Framework.
// #
// #   			The StrataCommon Framework is free software: you 
// #			can redistribute it and/or modify it under the terms of 
// #			the GNU Lesser General Public License as published by
// #    		the Free Software Foundation, either version 3 of the 
// #			License, or (at your option) any later version.
// #
// #    		The StrataCommon Framework is distributed in the 
// #			hope that it will be useful, but WITHOUT ANY WARRANTY; 
// #			without even the implied warranty of MERCHANTABILITY or 
// #			FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser 
// #			General Public License for more details.
// #
// #    		You should have received a copy of the GNU Lesser 
// #			General Public License along with the StrataCommon
// #			Framework. If not, see http://www.gnu.org/licenses/.
// ##########################################################################

package strata1.common.producerconsumer;

import com.lmax.disruptor.ExceptionHandler;

/****************************************************************************
 * 
 * @author 		
 *     Sapientia Systems
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public 
class DisruptorExceptionHandler
    implements ExceptionHandler
{
    private IExceptionHandler itsAdaptee;
    
    /************************************************************************
     * Creates a new {@code DisruptorExceptionHandler}. 
     *
     */
    public 
    DisruptorExceptionHandler(IExceptionHandler adaptee)
    {
        itsAdaptee = adaptee;
    }

    /************************************************************************
     * {@inheritDoc} 
     */
    @Override
    public void 
    handleEventException(Throwable ex,long sequence,Object event)
    {
        itsAdaptee.onException( ex );
    }

    /************************************************************************
     * {@inheritDoc} 
     */
    @Override
    public void 
    handleOnStartException(Throwable ex)
    {
        itsAdaptee.onException( ex );
    }

    /************************************************************************
     * {@inheritDoc} 
     */
    @Override
    public void 
    handleOnShutdownException(Throwable ex)
    {
        itsAdaptee.onException( ex );
    }

}

// ##########################################################################

File: legacy/csharp/components/Strata.Foundation/ProducerConsumer/DisruptorExceptionHandler.cs
//  ##########################################################################
//  # File Name: DisruptorExceptionHandler.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################

using System;

namespace Strata.Foundation.ProducerConsumer
{

    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// 
    /// </summary>
    ///  
    public
    class DisruptorExceptionHandler<T>:
        Disruptor.IExceptionHandler<T>
    {
        private IExceptionHandler itsAdaptee;

        public 
        DisruptorExceptionHandler(IExceptionHandler adaptee)
        {
            itsAdaptee = adaptee;
        }

        public void 
        HandleEventException(Exception ex,long sequence,T evt)
        {
            itsAdaptee.OnException(ex);
        }

        public void 
        HandleOnShutdownException(Exception ex)
        {
            itsAdaptee.OnException(ex);
        }

        public void 
        HandleOnStartException(Exception ex)
        {
            itsAdaptee.OnException(ex);
        }
    }
}

//  ##########################################################################
